ocrd/ocrd/decorators.py

```diff
@@ -46,9 +56,25 @@ def ocrd_cli_wrap_processor(processorClass, ocrd_tool=None, mets=None, working_d
             raise Exception(msg)
         resolver = Resolver()
         workspace = resolver.workspace_from_url(mets, working_dir)
-        # TODO once we implement 'overwrite' CLI option and mechanism, disable the
-        # `output_file_grp_ check by setting to False-y value if 'overwrite' is set
-        report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'])
+        if overwrite:
+            if 'output_file_grp' not in kwargs or not kwargs['output_file_grp']:
+                raise Exception("--overwrite requires --output-file-grp")
+            LOG.debug("Removing files because of --overwrite")
+            for grp in kwargs['output_file_grp'].split(','):
+                if 'page_id' in kwargs:
+                    for page_id in kwargs['page_id'].split(','):
+                        LOG.debug("Removing files in output file group %s with page ID %s", grp, page_id)
+                        for file in workspace.mets.find_files(pageId=page_id, fileGrp=grp):
+                            workspace.remove_file(file, force=True, keep_file=False)
+                else:
+                    LOG.debug("Removing all files in output file group %s ", grp)
+                    workspace.remove_file_group(grp, recursive=True, force=True, keep_files=False)
+        if overwrite and 'page_id' in kwargs:
+            # If 'overwrite' and 'page_id' were given, it is possible the output
+            # file group is not completely empty
+            report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], '')
+        else:
+            report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'])
```

Comment:
Just a thought: Even _without_ `--overwrite`, if I merely want to process some page(s), I should not be prohibited to do so if the output file group already exists, but not for the page(s) I am interested in.

Reply:
We could check whether, provided `--page_id` is set, the output file group contains files representing those pages. It's doable but the effect would be the same as with `--overwrite` (if the pages are not in the output file group, nothing is changed). But I like that `--overwrite` conveys that the user is aware that the output file group already exists and that processing could result in data loss.

Reply:
Exactly!

Reply:
"Exactly" as in 
  
  * "extend the functionality of the file group check to take `--page-id` into account and allow existing output file groups as long as they do not contain these specific pages", or
  * "force the user to acknowledge that his could potentially be a destructive action by providing the `--overwrite` flag, even if no files are actually overwritten, just an existing file group used as output"
?

Reply:
I don't see it as either-or. Yes, we should allow adding pages to existing file groups _without_ `--overwrite`. There's nothing destructive in that. The user should always have that safe option. By making the user always need `--overwrite`, we reduce its use to cases where the user _should_ pay attention.

Reply:
`WorkspaceValidator.check_file_grp` now accepts a `page_id` parameter. If provided, an exception is only raised if the output group already contains **AND** there are files in that output file group for that page/these pages.